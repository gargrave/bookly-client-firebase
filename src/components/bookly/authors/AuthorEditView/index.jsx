// @flow
import React from 'react';
import { bool, func, object, string } from 'prop-types';

import type { Author } from '../../../../constants/flowtypes';

import AuthorForm from '../AuthorForm';
import Card from '../../../common/Card';

type Props = {
  author: Author,
  errors: Author,
  onCancel: Function,
  onInputChange: Function,
  onSubmit: Function,
  submitDisabled?: boolean,
  topLevelError?: string,
};

function AuthorEditView({
  author,
  errors,
  onCancel,
  onInputChange,
  onSubmit,
  submitDisabled = false,
  topLevelError,
}: Props) {
  return (
    <div className="author-edit-view">
      <Card
        classes={['card--top-margin-med']}
        header={"Update Author"}
        hoverable={false}
      >
        <AuthorForm
          author={author}
          errors={errors}
          onCancel={onCancel}
          onInputChange={onInputChange}
          onSubmit={onSubmit}
          submitDisabled={submitDisabled}
          topLevelError={topLevelError}
        />
      </Card>
    </div>
  );
}

AuthorEditView.propTypes = {
  author: object.isRequired,
  errors: object.isRequired,
  onCancel: func.isRequired,
  onInputChange: func.isRequired,
  onSubmit: func.isRequired,
  submitDisabled: bool,
  topLevelError: string,
};

export default AuthorEditView;
