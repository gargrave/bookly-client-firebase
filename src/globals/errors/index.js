// @flow
import { parseError, parseFbError } from './apiErrorParser';
import { validationErrors } from './validationErrors';

export {
  parseError,
  parseFbError,
  validationErrors,
};
