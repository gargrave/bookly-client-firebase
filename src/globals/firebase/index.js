import firebase from 'firebase';
import 'firebase/firestore';

import firebaseConfig from '../../secrets/firebaseConfig.js';

let initialized = false;
let db;

if (!initialized) {
  firebase.initializeApp(firebaseConfig);
  db = firebase.firestore();
  initialized = true;
}

function docToObject(doc) {
  return {
    id: doc.id,
    ...doc.data(),
  };
}

function timestamp() {
  return firebase.firestore.FieldValue.serverTimestamp();
}

export {
  db,
  docToObject,
  timestamp,
};
