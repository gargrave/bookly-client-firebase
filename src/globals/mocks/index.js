import { authorMocks } from './data/authorMocks';
import { bookBucketMocks } from './data/bookBucketMocks';
import { bookMocks } from './data/bookMocks';
import { profileMocks } from './data/profileMocks';
import { userMocks } from './data/userMocks';

export {
  authorMocks,
  bookBucketMocks,
  bookMocks,
  profileMocks,
  userMocks,
};
