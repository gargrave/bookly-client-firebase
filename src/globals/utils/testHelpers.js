import React from 'react';
import { shallow } from 'enzyme';

export class ComponentBuilder {
  constructor(component, defaultProps) {
    this.component = component;
    this.defaultProps = defaultProps;
  }

  shallowGetComponent(extraProps = {}) {
    const props = {
      ...this.defaultProps,
      ...extraProps,
    };
    return shallow(<this.component {...props} />);
  }
}
