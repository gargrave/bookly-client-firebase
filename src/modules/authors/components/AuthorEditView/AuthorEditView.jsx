// @flow
import React from 'react'
import { bool, func, object, string } from 'prop-types'

import type { Author, AuthorErrors } from '../../flowtypes'

import Card from '../../../common/components/Card/Card'
import AuthorForm from '../AuthorForm/AuthorForm'

import styles from './AuthorEditView.css'

type Props = {
  author: Author,
  disabled: boolean,
  errors: AuthorErrors,
  onCancel: Function,
  onInputChange: Function,
  onSubmit: Function,
  submitDisabled?: boolean,
  topLevelError?: string,
}

const AuthorEditView = ({
  author,
  disabled,
  errors,
  onCancel,
  onInputChange,
  onSubmit,
  submitDisabled = false,
  topLevelError,
}: Props) => {
  return (
    <div className={styles.authorEditView}>
      <Card header="Update Author">
        <AuthorForm
          author={author}
          disabled={disabled}
          errors={errors}
          onCancel={onCancel}
          onInputChange={onInputChange}
          onSubmit={onSubmit}
          submitDisabled={submitDisabled}
          topLevelError={topLevelError}
        />
      </Card>
    </div>
  )
}

AuthorEditView.propTypes = {
  author: object.isRequired,
  disabled: bool,
  errors: object.isRequired,
  onCancel: func.isRequired,
  onInputChange: func.isRequired,
  onSubmit: func.isRequired,
  submitDisabled: bool,
  topLevelError: string,
}

export default AuthorEditView
