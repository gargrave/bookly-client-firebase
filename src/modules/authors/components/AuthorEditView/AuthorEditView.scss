@import '../../../../styles/_views';

.authorEditView {
  @include viewWrapper;
}
