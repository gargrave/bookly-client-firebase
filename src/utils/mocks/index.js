import authorMocks from './data/authors';
import userMocks from './data/users';

export {
  authorMocks,
  userMocks,
};
