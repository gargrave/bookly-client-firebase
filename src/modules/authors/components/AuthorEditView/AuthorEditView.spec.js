import { ComponentBuilder } from '../../../../utils/testHelpers';

import { authorMocks } from '../../../../globals/mocks/';

import AuthorForm from '../AuthorForm/AuthorForm';

import AuthorEditView from './AuthorEditView';

const defaultProps = {
  author: { ...authorMocks[0] },
  errors: {
    firstName: '',
    lastName: '',
  },
  onCancel: jest.fn(),
  onInputChange: jest.fn(),
  onSubmit: jest.fn(),
  submitDisabled: false,
  topLevelError: '',
};

const builder = new ComponentBuilder(
  AuthorEditView, defaultProps,
);

describe('AuthorEditView', () => {
  let component;

  it('matches the snapshot', () => {
    component = builder.shallowGetComponent();
    expect(component).toMatchSnapshot();
  });

  it('renders the AuthorForm component', () => {
    component = builder.shallowGetComponent();
    expect(component.find(AuthorForm)).toHaveLength(1);
  });
});
