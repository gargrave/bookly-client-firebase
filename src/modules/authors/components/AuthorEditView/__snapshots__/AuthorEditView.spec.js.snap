// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthorEditView matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AuthorEditView
    author={
      Object {
        "bookCount": 1,
        "created": 2017-07-01T01:16:43.881Z,
        "firstName": "Aldous",
        "id": "oiw223h3qndlksadffl3ksdf",
        "lastName": "Huxley",
        "updated": 2017-07-01T01:16:43.881Z,
      }
    }
    errors={
      Object {
        "firstName": "",
        "lastName": "",
      }
    }
    onCancel={[Function]}
    onInputChange={[Function]}
    onSubmit={[Function]}
    submitDisabled={false}
    topLevelError=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Card
        header="Update Author"
        hoverable={false}
      >
        <AuthorForm
          author={
            Object {
              "bookCount": 1,
              "created": 2017-07-01T01:16:43.881Z,
              "firstName": "Aldous",
              "id": "oiw223h3qndlksadffl3ksdf",
              "lastName": "Huxley",
              "updated": 2017-07-01T01:16:43.881Z,
            }
          }
          disabled={undefined}
          errors={
            Object {
              "firstName": "",
              "lastName": "",
            }
          }
          onCancel={[Function]}
          onInputChange={[Function]}
          onSubmit={[Function]}
          submitDisabled={false}
          topLevelError=""
        />
      </Card>,
      "className": "bookly-edit-view bookly-author-edit-view",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <AuthorForm
          author={
            Object {
              "bookCount": 1,
              "created": 2017-07-01T01:16:43.881Z,
              "firstName": "Aldous",
              "id": "oiw223h3qndlksadffl3ksdf",
              "lastName": "Huxley",
              "updated": 2017-07-01T01:16:43.881Z,
            }
          }
          disabled={undefined}
          errors={
            Object {
              "firstName": "",
              "lastName": "",
            }
          }
          onCancel={[Function]}
          onInputChange={[Function]}
          onSubmit={[Function]}
          submitDisabled={false}
          topLevelError=""
        />,
        "header": "Update Author",
        "hoverable": false,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "author": Object {
            "bookCount": 1,
            "created": 2017-07-01T01:16:43.881Z,
            "firstName": "Aldous",
            "id": "oiw223h3qndlksadffl3ksdf",
            "lastName": "Huxley",
            "updated": 2017-07-01T01:16:43.881Z,
          },
          "disabled": undefined,
          "errors": Object {
            "firstName": "",
            "lastName": "",
          },
          "onCancel": [Function],
          "onInputChange": [Function],
          "onSubmit": [Function],
          "submitDisabled": false,
          "topLevelError": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Card
          header="Update Author"
          hoverable={false}
        >
          <AuthorForm
            author={
              Object {
                "bookCount": 1,
                "created": 2017-07-01T01:16:43.881Z,
                "firstName": "Aldous",
                "id": "oiw223h3qndlksadffl3ksdf",
                "lastName": "Huxley",
                "updated": 2017-07-01T01:16:43.881Z,
              }
            }
            disabled={undefined}
            errors={
              Object {
                "firstName": "",
                "lastName": "",
              }
            }
            onCancel={[Function]}
            onInputChange={[Function]}
            onSubmit={[Function]}
            submitDisabled={false}
            topLevelError=""
          />
        </Card>,
        "className": "bookly-edit-view bookly-author-edit-view",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <AuthorForm
            author={
              Object {
                "bookCount": 1,
                "created": 2017-07-01T01:16:43.881Z,
                "firstName": "Aldous",
                "id": "oiw223h3qndlksadffl3ksdf",
                "lastName": "Huxley",
                "updated": 2017-07-01T01:16:43.881Z,
              }
            }
            disabled={undefined}
            errors={
              Object {
                "firstName": "",
                "lastName": "",
              }
            }
            onCancel={[Function]}
            onInputChange={[Function]}
            onSubmit={[Function]}
            submitDisabled={false}
            topLevelError=""
          />,
          "header": "Update Author",
          "hoverable": false,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "author": Object {
              "bookCount": 1,
              "created": 2017-07-01T01:16:43.881Z,
              "firstName": "Aldous",
              "id": "oiw223h3qndlksadffl3ksdf",
              "lastName": "Huxley",
              "updated": 2017-07-01T01:16:43.881Z,
            },
            "disabled": undefined,
            "errors": Object {
              "firstName": "",
              "lastName": "",
            },
            "onCancel": [Function],
            "onInputChange": [Function],
            "onSubmit": [Function],
            "submitDisabled": false,
            "topLevelError": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
